src/items/modules.md

```diff
@@ -40,21 +40,26 @@ struct, enumeration, union, type parameter or crate can't shadow the name of a
 module in scope, or vice versa. Items brought into scope with `use` also have
 this restriction.
 
-A module without a body is loaded from an external file, by default with the
-same name as the module, plus the `.rs` extension. When a nested submodule is
-loaded from an external file, it is loaded from a subdirectory path that
-mirrors the module hierarchy.
+A module without a body is loaded from an external file. By default, the path
+to the file mirrors the logical [module path]. Ancestor path components are
+directories, and the module's contents are in a file with the name of the
+module plus the `.rs` extension. For example, the following module structure
+can have this corresponding filesystem structure:
 
-```rust,ignore
-// Load the `vec` module from `vec.rs`
-mod vec;
+Module Path           | Filesystem Path  | File Contents
+--------------------- | ---------------  | -------------
+`crate`               | `lib.rs`         | `mod util;`
+`crate::util`         | `util.rs`        | `mod config;`
+`crate::util::config` | `util/config.rs` |
 
-mod thread {
-    // Load the `local_data` module from `thread/local_data.rs`
-    // or `thread/local_data/mod.rs`.
-    mod local_data;
-}
-```
+> **Note**: Module filenames may also be the name of the module as a directory
```

Comment:
That `util/mod.rs` is also valid is a part of the language, and not just a note. The previous to Rust 1.30 should stay as a note (though I'd rather it say `rustc`, since I don't think Rust is versioned right now). We should also explicitly say it is deprecated.

---

Also, is this allowed?

```rust
#[path = "util.rs"] mod util_1;
#[path = "util/mod.rs"] mod util_2;
```

Reply:
I'll move it out of the note, though that may complicate it some.

> We should also explicitly say it is deprecated.

I wouldn't say it is deprecated.  The RFC explicitly says "The use of mod.rs continues to be allowed without any deprecation."  To me, deprecated usually implies a lint that triggers, but in this case not even the edition-idiom lints recommend this. I carefully chose to say it is "encouraged" though I don't think anyone has officially established even that.

> Also, is this allowed?

Haha, yes.

Reply:
Alright. Scratch the deprecation part then.

Though, given the interaction exists, I'm thinking it might it make sense to describe the chosen file as an algorithm? I want to say probably not, since we can statically state that the path attribute changes the resolution. But in that case, we should state that this is only the case when the path attribute is not applied.

Reply:
> this is only the case when the path attribute is not applied.

Yea, I tried to imply that by starting with "By default", but that is a little too subtle.  I'll try to think of a way to make it clear the distinction between "default" behavior and overridden behavior.

Reply:
Given that the only non-default is when there's a path attribute, you can explicitly say "When the module has no path attribute"